pinot-common/src/main/java/com/linkedin/pinot/common/utils/request/RequestUtils.java

```diff
@@ -149,18 +151,58 @@ public static FilterQueryTree buildFilterQuery(Integer id, Map<Integer, FilterQu
   }
 
   /**
-   * Helper method to extract all column names from group-by expressions.
+   * Extracts all columns from the given filter query tree.
    */
-  public static Set<String> getAllGroupByColumns(@Nullable GroupBy groupBy) {
-    Set<String> allGroupByColumns = new HashSet<>();
-
-    if (groupBy != null) {
-      for (String expression : groupBy.getExpressions()) {
-        TransformExpressionTree expressionTree = TransformExpressionTree.compileToExpressionTree(expression);
-        expressionTree.getColumns(allGroupByColumns);
+  @Nonnull
+  public static Set<String> extractFilterColumns(@Nonnull FilterQueryTree root) {
+    Set<String> filterColumns = new HashSet<>();
+    if (root.getChildren() == null) {
+      filterColumns.add(root.getColumn());
+    } else {
+      Stack<FilterQueryTree> stack = new Stack<>();
+      stack.add(root);
+      while (!stack.empty()) {
+        FilterQueryTree node = stack.pop();
+        for (FilterQueryTree child : node.getChildren()) {
+          if (child.getChildren() == null) {
```

Comment:
check for both null and empty?

Reply:
**children** is non-null only for AND and OR, and will never be empty